### DISCUSSION TITLE
[AngularJS] Rotas com parametros

### DISCUSSION THREAD

**Discussion by gustavofp** [frontendbr/forum#605] [Perguntas e Respostas]
Como funciona uma aplicação com rotas dinâmicas no Angular?
Não tenho muita experiência, porém do pouco que utilizei tive problemas pra entender esse negócio de parametro nas rotas.
Por exemplo, eu tenho uma rota que é 'users/:id', o correto é eu abrir essa rota colocando o Id como parâmetro e a página carregar com as infos do usuário. Porém quando tento acessar essa URL direto, ele sempre retorna 'Cannot / GET users'
Como consigo contornar esse problema?

**Comment by felipefialho** [comment]
@gustavofp ng2? Se sim, ficaria algo como

```
this.activatedRoute.params.forEach((params: Params) => {
   const id = params['id']; 
   ...
});
```
